client/components/product-details.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+class ProductDetails extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      product: {}
```

Comment:
It's better to start this as `null` and then just render `null` until the state updates with the product. It's too risky to render an object with a bunch of undefined properties.